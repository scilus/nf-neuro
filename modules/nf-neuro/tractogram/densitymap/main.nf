process TRACTOGRAM_DENSITYMAP {
    tag "$meta.id"
    label 'process_single'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://scil.usherbrooke.ca/containers/scilus_2.0.2.sif':
        'scilus/scilus:2.0.2' }"

    input:
    tuple val(meta), path(tractogram)

    output:
    tuple val(meta), path("*__*.nii.gz"), emit: density_map
    path "versions.yml"                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def binary = task.ext.is_binary ? "--binary" : ""
    """
    bname=\$(basename ${tractogram} .trk)
    scil_tractogram_compute_density_map.py $tractogram ${prefix}__\${bname}.nii.gz ${binary}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: \$(pip list --disable-pip-version-check --no-python-version-warning | grep scilpy | tr -s ' ' | cut -d' ' -f2)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    scil_tractogram_compute_density_map.py -h

    touch ${prefix}_AF_L.nii.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: \$(pip list --disable-pip-version-check --no-python-version-warning | grep scilpy | tr -s ' ' | cut -d' ' -f2)
    END_VERSIONS
    """
}
