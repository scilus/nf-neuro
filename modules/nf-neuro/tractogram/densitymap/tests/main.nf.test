nextflow_process {

    name "Test Process TRACTOGRAM_DENSITYMAP"
    script "../main.nf"
    process "TRACTOGRAM_DENSITYMAP"

    tag "modules"
    tag "modules_nfcore"
    tag "tractogram"
    tag "tractogram/densitymap"

    tag "subworkflows"
    tag "subworkflows/load_test_data"

    setup {
        run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
            script "../../../../../subworkflows/nf-neuro/load_test_data/main.nf"
            process {
                """
                input[0] = Channel.from( [ "bundles.zip" ] )
                input[1] = "test.load-test-data"
                """
            }
        }
    }

    test("densitymap - map") {

        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/bundle_all_1mm.trk")
                    ]
                }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("densitymap - mask") {

        config "./nextflow_mask.config"

        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/bundle_all_1mm.trk")
                    ]
                }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}
