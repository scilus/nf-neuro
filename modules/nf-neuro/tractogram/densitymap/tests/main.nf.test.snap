{
    "densitymap - map": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__bundle_all_1mm.nii.gz:md5,a9cc8ba016a680ec8cbac940c45e5464"
                    ]
                ],
                "1": [
                    "versions.yml:md5,7a2b58d5ff9231cb698c8ca1a279171f"
                ],
                "density_map": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__bundle_all_1mm.nii.gz:md5,a9cc8ba016a680ec8cbac940c45e5464"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7a2b58d5ff9231cb698c8ca1a279171f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-10-31T13:26:09.344942322"
    },
    "densitymap - mask": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__bundle_all_1mm.nii.gz:md5,5c682d295c593e132295e21e753a6f4f"
                    ]
                ],
                "1": [
                    "versions.yml:md5,7a2b58d5ff9231cb698c8ca1a279171f"
                ],
                "density_map": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__bundle_all_1mm.nii.gz:md5,5c682d295c593e132295e21e753a6f4f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7a2b58d5ff9231cb698c8ca1a279171f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-10-31T13:26:15.496099088"
    }
}